src/Amadevus.RecordGenerator.Generators/GeneratedCodeAttributeGenerator.cs

```diff
@@ -12,32 +12,96 @@ namespace Amadevus.RecordGenerator.Generators
+            },
+            {
+                typeof(ClassDeclarationSyntax),
+                (context) => context.AttributeApplier.GetWithGeneratedCodeAttributeOnMembers((ClassDeclarationSyntax)context.Member)
+            }
+        };
+
+
+        public GeneratedCodeAttributeApplier(
+            GeneratedCodeAttributeGenerator attributeGenerator)
+        {
+            this.attributeGenerator = attributeGenerator;
+        }
 
-        public GeneratedCodeAttributeGenerator(
-            TypeDeclarationSyntax typeSyntax)
+        
+        public ClassDeclarationSyntax GetWithGeneratedCodeAttributeOnMembers(
+            ClassDeclarationSyntax typeDeclaration)
         {
-            this.typeSyntax = typeSyntax;
+            Func<SyntaxNode, bool> toBeReplaced = (node) =>
+            {
+                return Updates.Keys.Any(k => k.IsAssignableFrom(node.GetType()));
+            };
+
+            Func<SyntaxNode, SyntaxNode, SyntaxNode> calculateReplacement = (rootNode, toBeReplacedNode) =>
+            {
+                var context = new MemberReplacementContext(toBeReplacedNode, this, attributeGenerator);
+                var updater = Updates[toBeReplacedNode.GetType()];
+                return updater(context);
+            };
+
+            return typeDeclaration.ReplaceNodes(
+                typeDeclaration.ChildNodes().Where(toBeReplaced),
+                calculateReplacement);
         }
 
-        public TypeDeclarationSyntax GetWithGeneratedCodeAttribute()
+        private class MemberReplacementContext
+        {
+            public SyntaxNode Member { get; }
+            public GeneratedCodeAttributeApplier AttributeApplier { get; }
+            public GeneratedCodeAttributeGenerator AttributeGenerator { get; }
+
+            public MemberReplacementContext(
+                SyntaxNode member,
+                GeneratedCodeAttributeApplier attributeApplier,
+                GeneratedCodeAttributeGenerator attributeGenerator)
+            {
+                Member = member;
+                AttributeApplier = attributeApplier;
+                AttributeGenerator = attributeGenerator;
+            }
+        }
+    }
+
+    internal sealed class GeneratedCodeAttributeGenerator
+    {
+        public SyntaxList<AttributeListSyntax> GenerateAttributeListSyntaxList()
         {
-            return typeSyntax.WithAttributeLists(
-                SingletonList(
-                    AttributeList(
-                        SingletonSeparatedList(
-                            Attribute(GenerateQualifiedName())
-                            .WithArgumentList(GenerateAttributeArgumentList())))));
+            return SingletonList(
+                AttributeList(
+                    SingletonSeparatedList(
+                        Attribute(GenerateQualifiedName())
+                            .WithArgumentList(GenerateAttributeArgumentList()))));
```

Comment:
The indentation here is off (it need to be decreased for the `WithArgumentsList`.